client/client.go

```diff
@@ -490,7 +490,7 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		case ctx.Err() != nil:
 			err = ctx.Err()
 		case hctx.Err() != nil:
-			err = fmt.Errorf("client: endpoint %s exceeded header timeout", c.endpoint.String())
+			err = hctx.Err()
```

Comment:
I see. So it's expected to return `client: etcd cluster is unavailable or misconfigure` rather than context error? I can close this and the issue https://github.com/coreos/etcd/issues/4544.